README.md

```diff
@@ -0,0 +1,139 @@
+        }, {}, import("mobx-state-tree")._NotCustomized, import("mobx-state-tree")._NotCustomized>>;
+    }, {}, import("mobx-state-tree")._NotCustomized, import("mobx-state-tree")._NotCustomized>>, [undefined]>;
+}, {}, import("mobx-state-tree")._NotCustomized, import("mobx-state-tree")._NotCustomized>;
+```
+What we can see here is that instead of referencing types of the child models, we get full duplication/inlining of child models declarations.
+That happens for each level inside the stores tree will have the same duplication up to it's level.
+
+
+## ...But it's not impact only types emit!
+
+You can think of types emit as visualization of the inferred data structures inside typescript language server. duplication in `d.ts` = more work `tsc/tsserver` need to do or regular basis.  
+And indeed when working with big and nested models, the build time and IDE responsiveness of typescript types is badly degraded
+
+## Peeling the onion 🧅
+
+On the coming articles, we will cover issues with `MST` and workarounds for them, one by one, like peeling a fire roasted onion to get to the tasty parts!
+
+## Layer one: reduce import() references
+When examining the emitted types for `Store1` above, the first thing that catches my eye is the many references to `import("mobx-state-tree")` for the same symbol. 
+`import("mobx-state-tree").IModelType` appears 5 times!
+
+What can we do about it? Lets import the types directly!  
+Adding `import type * as MST from "mobx-state-tree";` to our model file will turn the emitted types into:
+```ts
+import type * as MST from "mobx-state-tree";
+export declare const Store1: MST.IModelType<{
+    store2: MST.IOptionalIType<MST.IArrayType<MST.IModelType<{
+        store3: MST.IArrayType<MST.IModelType<{
+            store4: MST.IModelType<{
+                store5: MST.IModelType<{
+                    prop1: MST.IMaybe<MST.ISimpleType<string>>;
+                }, {}, MST._NotCustomized, MST._NotCustomized>;
+            }, {}, MST._NotCustomized, MST._NotCustomized>;
+            store5: MST.IOptionalIType<MST.IModelType<{
+                prop1: MST.IMaybe<MST.ISimpleType<string>>;
+            }, {}, MST._NotCustomized, MST._NotCustomized>, [undefined]>;
+        }, {}, MST._NotCustomized, MST._NotCustomized>>;
+    }, {}, MST._NotCustomized, MST._NotCustomized>>, [undefined]>;
+}, {}, MST._NotCustomized, MST._NotCustomized>;
+
+```
+
+Much more compact and readable!  
+You may also import individual symbols instead of a star import
+
+## Layer two: Child stores inlining
+
+The second thing that catches the eye is that, even thu `Store2` (And the rest) are declared on a separate file, we ended up with the entire stores types inline inside `Store1`.
+
+Let's understand why that's happing:
+When typescript does not have a clear name for type information, it will be inlined into his reference points!  
```

Comment:
```suggestion
When TypeScript does not have a clear name for type information, it will be inlined into its reference points!  
```

Reply:
What are `reference points` exactly? If that's a thing which you didn't invent, add a link as a reference.